### DISCUSSION TITLE
Post-Recording Analysis with BirdNET

### DISCUSSION THREAD

**Discussion by Vollpflock** [birdnet-team/BirdNET-Analyzer#233] [Q&A] [answered]
_I noticed Stefan enabled the Discussions tab, so rather than spending a long time searching, I decided to directly ask here. Please correct me if I'm misunderstanding something._

I have very little experience with BirdNET-Analyzer. I've only been using BirdNET-Pi and tried BirdNETlib briefly on my Mac. 

I have recorded thousands of hours of audio at remote locations that need to be analyzed. The files are each 1 hour long and in the format date-time.wav. 

That's why I'm looking for **a solution on how to analyze audio data post-recording.** I'm a birder and my main goal is to identify species. So, it's crucial for me to display spectrograms and play back sounds.

Can someone help me out here? Maybe I'm missing something, but BirdNET-Analyzer doesn't seem to be designed to display spectrograms and play back sounds in large quantity

And I really like BirdNET-Pi, especially since it displays the recordings by date and shows spectrograms, but it appears to be for real-time analysis only.

Other projects and ideas I've read about mainly focus on battery-powered, real-time analysis, which is great for live streaming. However, for my needs, recording the audio first and then analyzing it later seems more straightforward and sufficient. This method also appears to be more energy-efficient, especially for remote, battery-powered solutions.

**Comment by kahst** [comment, marked as answer]
You can probably approach this in two steps: First, run your data through the analyzer and select a result format, which will then allow you to view the results with spectrograms in a second step. We support [Raven](https://www.birds.cornell.edu/ccb/raven-website-down-interim-access-to-raven-resources/) and [Audacity](https://www.audacity.de), both programs allow you to review the detections. We are working on a review workflow for the Analyzer, but for now I think it's easiest to use third-party software.

**Reply by crywolfproject** [reply]
Are you dropping support for Kaleidoscope? Many of our wildlife managers prefer its ability to load the results.csv because Kaleidoscope automatically loads all associated clips and it is very easy to tab through and validate detections. It would be fine if BirdNet eventually has its own workflow but as long as it is a very efficient workflow, since the agency folks are really pushing the "time is money" mantra.

**Reply by kahst** [reply]
No, we will continue to support other software, I only mentioned Raven and Audacity above for the sake of brevity. In fact, if you think we should add another exchange or metadata format, let us know and we'll give it a try.

**Reply by Vollpflock** [reply]
Thanks, Stefan. Yes, I have done that, but it hasn't worked effectively for me for analyzing large amounts of audio files. Not sure if I’m doing something wrong, though. I’ll try again and let you know.

**Comment by Mattk70** [comment]
Hi @Vollpflock ,

I have spent the last couple of years developing a Windows desktop application for Nocmig recording analysis. The use case is very similar to yours. I too have thousands of hours of night time recordings which I analyse after the fact. Chirpity, the application, loads up audio files and processes them using a model I trained myself. Given the duration of the recordings I have spent a lot of time optimising the inference speed - the time it takes to analyse the audio. I'm pleaesd to say that it is very fast, and the accuracy is comparable to BirdNET. Moreover, you can drop a folder of audio and set it running. In my case, it will chug though 8 hours of audio in just over 30 seconds (although you will need a top notch graphics card to emulate this performance!).

Whilst my training data is limited to the British list, so it will not detect American species (for example) with @kahst 's help I have just got the latest BirdNET model plugged into the app. This means it can identify any species BirdNET knows about!

Chirpity doesn't *just* identify species. It displays a spectrogram of the audio, and selecting any detection will jump you straight to the part of the audio file where that detection was encountered. It will also allow you to play the call and annotate / amend detections. Finally, you can save your records to a database for future reference - or export the detections as a CSV, Audacity label file, or even an audio clip in your chosen format.

The curent release doesn't have the latest BirdNET model intagration (that will come very soon) , but you can read more about it here: https://chirpity.mattkirkland.co.uk

**Reply by crywolfproject** [reply]
Well done on the UI. Let me know when you have the latest BirdNET integration and I'll spread the word in my circles.

**Reply by Vollpflock** [reply]
Hey @Mattk70 

I need to install Windows on my Mac first, but this seems to be what I've been craving. Thank you so much! I'll try it out and let you know how it goes. Could you may open the “Discussions”-tab in your github?

**Reply by Mattk70** [reply]
Hi @Vollpflock - sure, I've just done that

**Reply by Mattk70** [reply]
@reedflyfarm I have now released a version of Chirpity with the latest BirdNET integration
